File: python/pick6.py
import random
import string

def pick_six():

    numbers = []
    counter = 0

    while counter < 6:

        x = random.randint(1,99)
        numbers.append(x)
        counter += 1
    
    return numbers

def main():

    print('Welcome to the Pick 6 Lotto.')
    
    ticket_cost = 0
    total_winnings = 0
    match_1 = 0
    match_2 = 0
    match_3 = 0
    match_4 = 0
    match_5 = 0
    match_6 = 0
    winning_ticket = pick_six()
    winning_ticket.sort()
    
    game_runs = input('How many picks would you like to try?: ')

    x = 0
    
    while x < int(game_runs):

        user_ticket = pick_six()
        user_ticket.sort()
        
        ticket_cost += 2
        
        match_count = 0
        
        winning_amount = 0

        for ele in user_ticket:

            if ele in winning_ticket:
                match_count += 1
        
        if match_count == 1:
            winning_amount = 4
            match_1 += 1
        
        if match_count == 2:
            winning_amount = 7
            match_2 += 1
        
        if match_count == 3:
            winning_amount = 100
            match_3 += 1
        
        if match_count == 4:
            winning_amount = 50000
            match_4 += 1
        
        if match_count == 5:
            winning_amount = 1000000
            match_5 += 1
        
        elif match_count == 6:
            winning_amount = 25000000
            match_6 += 1

        x += 1
        total_winnings += winning_amount            
        
        print(user_ticket)
        print(winning_ticket)
        print(match_count)

    print(f'\nYou\'re winning total amount is: {total_winnings}')
    print(f'Your\'re total ticket cost is: {ticket_cost}')
    print(f'You have matched 1 number {match_1} times.')
    print(f'You have matched 2 number {match_2} times.')
    print(f'You have matched 3 number {match_3} times.')
    print(f'You have matched 4 number {match_4} times.')
    print(f'You have matched 5 number {match_5} times.')
    print(f'You have matched 6 number {match_6} times.')
    
main()

File: javascript/pick6/pick6.js
function pickSix(arrayName){
    for (let i = 0; i < arrayName.length; i++){
        arrayName[i] = Math.floor(Math.random()*100)
    };
};

function ticketCompare(arrayName1, arrayName2, tickets){

    let totalWinnings = 0;

    for(let run = 0; run < tickets; run++){
        pickSix(arrayName2);
        console.log(arrayName1);
        console.log(arrayName2)
        console.log('\n')
        let matchCount = 0;
        let winningAmount = 0;
    
        for (let i=0; i < arrayName1.length; i++){
            if (arrayName1[i] === arrayName2[i]){
                matchCount +=1
                console.log(matchCount)
            };
        };
    
        switch(matchCount){
            case 1:
                winningAmount = 4;
                break;
            
            case 2:
                winningAmount = 7;
                break;
            
            case 3:
                winningAmount = 100;
                break;
    
            case 4:
                winningAmount = 50000;
                break;
            
            case 5:
                winningAmount = 1000000;
                break;
            
            case 6:
                winningAmount = 25000000;
                break;

            default:
                winningAmount = 0;
                break;
            }
       totalWinnings += winningAmount;
    };
    return(totalWinnings)
};

function roi(winnings,costs){
    return[(winnings - costs)/costs];
};

function main(){

    console.log('Welcome to Pick 6.','\n', 'Winning picks are the following:','\n', '1 number matched wins $4.','\n', '2 numbers matched wins $7.','\n', '3 numbers matched wins $100.', '\n', '4 numbers matched wins $50,000.', '\n', '5 numbers matched wins $1,000,000.','\n', '6 numbers matched wins $25,000,000.');

    let ticketAmount = 100;
    let totalTicketCost = (ticketAmount * 2);
    let winningNumbers = [0,0,0,0,0,0];
    let userTicket = [0,0,0,0,0,0];

    let wnArrayLength = 6;
    pickSix(winningNumbers);

    balance = ticketCompare(winningNumbers, userTicket, ticketAmount,) 

    console.log('You have won $', balance,'\n','You have spent $',totalTicketCost,'\n',
                'And your return on investment is ', roi(balance,totalTicketCost),'%.');
};

main()